docs/API.rst

```diff
@@ -0,0 +1,83 @@
+API for ContestWebServer
+************************
+
+Authentication
+==============
+
+All the log-in methods and restrictions that apply to normal ContestWebServer
+usage also apply for API usage. In particular, this implies that IP autologin,
+if configured, can be used for API calls.
+
+Otherwise, making a `POST` request to the `/api/login` endpoint, passing
+`username` and `password` as form data (as is done for regular login), will
+return a JSON object with the following structure:
+
+.. sourcecode:: json
+
+  {"login_data": string}
+
```

Comment:
We should specify what happens on errors (preferably once for the whole API).

Reply:
Done.